### ISSUE TITLE
AttributeError: 'function' object has no attribute 'get'

### ISSUE THREAD

**Issue by petrovicboban** [Crapworks/RESTlos#5]
`$ curl -H "content-type: application/json" 'http://admin:password@x.x.x.x/host?host_name=app'` returns  
`curl: (52) Empty reply from server`

Application crashes with:
```
\# ./restlosapi.py   


 \* starting restlos V0.3
 \* Running on http://0.0.0.0:80/ (Press CTRL+C to quit)
91.222.7.8 - - [12/Aug/2016 12:10:13] "GET /host HTTP/1.1" 200 -
Traceback (most recent call last):
  File "./restlosapi.py", line 363, in <module>
    app.run(host='0.0.0.0', port=config['port'])
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 843, in run
    run_simple(host, port, self, **options)
  File "/usr/lib/python2.7/site-packages/werkzeug/serving.py", line 694, in run_simple
    inner()
  File "/usr/lib/python2.7/site-packages/werkzeug/serving.py", line 659, in inner
    srv.serve_forever()
  File "/usr/lib/python2.7/site-packages/werkzeug/serving.py", line 499, in serve_forever
    HTTPServer.serve_forever(self)
  File "/usr/lib64/python2.7/SocketServer.py", line 238, in serve_forever
    self._handle_request_noblock()
  File "/usr/lib64/python2.7/SocketServer.py", line 297, in _handle_request_noblock
    self.handle_error(request, client_address)
  File "/usr/lib64/python2.7/SocketServer.py", line 295, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/usr/lib64/python2.7/SocketServer.py", line 321, in process_request
    self.finish_request(request, client_address)
  File "/usr/lib64/python2.7/SocketServer.py", line 334, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/lib64/python2.7/SocketServer.py", line 649, in __init__
    self.handle()
  File "/usr/lib/python2.7/site-packages/werkzeug/serving.py", line 216, in handle
    rv = BaseHTTPRequestHandler.handle(self)
  File "/usr/lib64/python2.7/BaseHTTPServer.py", line 340, in handle
    self.handle_one_request()
  File "/usr/lib/python2.7/site-packages/werkzeug/serving.py", line 251, in handle_one_request
    return self.run_wsgi()
  File "/usr/lib/python2.7/site-packages/werkzeug/serving.py", line 193, in run_wsgi
    execute(self.server.app)
  File "/usr/lib/python2.7/site-packages/werkzeug/serving.py", line 181, in execute
    application_iter = app(environ, start_response)
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 2000, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1991, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1559, in handle_exception
    handler = self._find_error_handler(InternalServerError())
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1476, in _find_error_handler
    .get(code))
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1465, in find_handler
    handler = handler_map.get(cls)
AttributeError: 'function' object has no attribute 'get'
```

It''s python 2.7.5, and Flask 0.11.1
This also happens when I try to create some hosts.

**Comment by Crapworks** [comment]
Hi @petrovicboban,

Thanks for your report!
there was a change in Flask 0.11 that prevents my method of adding error handler to work properly. But anyways, my method was not the right way to do it. Can you please check out the development branch and tell me if the error is gone? cba4419104f62f4f3a7fa9a8f1cb0bf270ec95e5 Should fix your issue.

If you tell me it workend, I will merge it back to master.

Regards,
Christian

**Comment by petrovicboban** [comment]
Hi,
there is no previous error, but the new one arised:

\# ./restlosapi.py 
Traceback (most recent call last):
  File "./restlosapi.py", line 383, in <module>
    app = NagiosAPI(\__name\__)
  File "./restlosapi.py", line 343, in \__init\__
    self._register_error_handler()
  File "./restlosapi.py", line 374, in _register_error_handler
    self.errorhandler(code)(self._error_handler)
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1175, in decorator
    self._register_error_handler(None, code_or_exception, f)
TypeError: _register_error_handler() takes exactly 1 argument (4 given)

**Comment by Crapworks** [comment]
Hi @petrovicboban,

obviously they also used the same name for class methods than I did... I just renamed them in the development branch. Can you check it out and try again? Sorry but I don't have access to a python installation so I am working blind right now :)

Regards,
Christian

**Comment by petrovicboban** [comment]
Hi @Crapworks ,
Thanks for efforts.
This is what I get:
{
  "code": 500, 
  "message": "500: Internal Server Error"
}


and this is how application crashes:

\# ./restlosapi.py 
 \* starting restlos V0.3
 \* Running on http://0.0.0.0:80/ (Press CTRL+C to quit)
[2016-08-12 17:10:15,780] ERROR in app: Exception on /host [GET]
Traceback (most recent call last):
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1988, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1641, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1544, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1639, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1625, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/opt/RESTlos/utils/authentication/__init__.py", line 72, in wrapped_function
    return f(*args, **kwargs)
  File "/usr/lib64/python2.7/site-packages/flask/views.py", line 84, in view
    return self.dispatch_request(*args, **kwargs)
  File "/usr/lib64/python2.7/site-packages/flask/views.py", line 149, in dispatch_request
    return meth(*args, **kwargs)
  File "./restlosapi.py", line 222, in get
    validate = self.endpoints.validate(self.endpoint, request.args)
  File "./restlosapi.py", line 91, in validate
    for illegal_char in self.main_cfg_values['illegal_object_name_chars']:
KeyError: 'illegal_object_name_chars'
Exception on /host [GET]
Traceback (most recent call last):
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1988, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1641, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1544, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1639, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1625, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/opt/RESTlos/utils/authentication/__init__.py", line 72, in wrapped_function
    return f(*args, **kwargs)
  File "/usr/lib64/python2.7/site-packages/flask/views.py", line 84, in view
    return self.dispatch_request(*args, **kwargs)
  File "/usr/lib64/python2.7/site-packages/flask/views.py", line 149, in dispatch_request
    return meth(*args, **kwargs)
  File "./restlosapi.py", line 222, in get
    validate = self.endpoints.validate(self.endpoint, request.args)
  File "./restlosapi.py", line 91, in validate
    for illegal_char in self.main_cfg_values['illegal_object_name_chars']:
KeyError: 'illegal_object_name_chars'
91.185.99.193 - - [12/Aug/2016 17:10:15] "GET /host?host_name=app HTTP/1.1" 500 -

**Comment by petrovicboban** [comment]
The last problem happened because nagios main cfg file missing illegal_object_name_chars parameter. I've created  pull request for fix.

**Comment by Crapworks** [comment]
Thanks for the pull request. I would liket to fix it not with a except/pass construction, but with providing a default: f3799c5706a741b0e70d4ba66d445fadd3bb493c
Can you please modify the pull request to only include the host binding? Then I can merge it.

**Comment by petrovicboban** [comment]
Hi, it's done.

**Comment by Crapworks** [comment]
Just merged #6 
Can you confirm that everything works now, so I can merge the development branch to master?

**Comment by petrovicboban** [comment]
It's working now. I will close this issue.
